docs/contributing/Developing a Language Feature.md

```diff
@@ -0,0 +1,67 @@
+# Developing a Language Feature 
+
+Adding a new feature to C# or VB is a very serious undertaking that often takes several iterations to complete for even the (seemingly) simplest of features. This is due to both the inherent complexity of changing languages and the need to consider the effects of new features in all layers of the Roslyn codebase: IDE, debugging, scripting, etc. As such, language work occurs in a separate branch until the feature reaches a point when we are ready to merge it into the main compiler.
+
+This page discusses the process by which langauge featuere *implementations* are considered, prototyped, and fully accepted into the language.  This process is intended to be used by the compiler team and community.  
+
+## Process 
+
+1. **Feature specification filed**:  The speclet should be filed as a GitHub issue and contain:
+
+    a. A description of the feature (including any syntax changes involved)
+    a. Discussions about impacted areas, such as overload resolution and type inference.
```

Comment:
I don't think there can be an exhaustive list of "impacted areas", but I think a second sentence about the complexity of impacted areas for new features is warranted. Maybe something along the lines of "Think through the major areas of the language specification while determine the potential impacts."